lib/src/reactive_ble.dart

```diff
@@ -28,41 +29,22 @@ class FlutterReactiveBle {
     _trackStatus();
   }
 
-  final _methodChannel = const MethodChannel("flutter_reactive_ble_method");
-
+  /// Registry that keeps track of all BLE devices found during BLE scan.
   final scanRegistry = DiscoveredDevicesRegistry();
 
-  BleStatus _status = BleStatus.unknown;
-
-  final Stream<BleStatus> _statusStream =
-      const EventChannel("flutter_reactive_ble_status")
-          .receiveBroadcastStream()
-          .cast<List<int>>()
-          .map((data) => pb.BleStatusInfo.fromBuffer(data))
-          .map(const ProtobufConverter().bleStatusFrom);
-
-  PrescanConnector _prescanConnector;
-
-  BleStatus get status => _status;
-
+  /// A stream that provides information about the status of BLE on the device.
+  ///
+  /// Emits the current [BleStatus] on subscription.
   Stream<BleStatus> get statusStream => Repeater(onListenEmitFrom: () async* {
         await initialize();
         yield _status;
         yield* _statusStream;
       }).stream;
 
-  Future<void> _trackStatus() async {
-    await initialize();
-    _statusStream.listen((status) => _status = status);
-  }
-
-  final Stream<ScanResult> _scanStream =
-      const EventChannel("flutter_reactive_ble_scan")
-          .receiveBroadcastStream()
-          .cast<List<int>>()
-          .map((data) => pb.DeviceScanInfo.fromBuffer(data))
-          .map(const ProtobufConverter().scanResultFrom);
+  /// Returns the current [BleStatus] of the device.
```

Comment:
The use of the word "device" is ambiguous: is it about a BLE peripheral or the host BLE central? We should be careful with this.